cmd/state-svc/manager.go

```diff
@@ -19,44 +19,53 @@ import (
 )
 
 type serviceManager struct {
-	cfg    *config.Instance
-	lockFp string
+	cfg *config.Instance
 }
 
 func NewServiceManager(cfg *config.Instance) *serviceManager {
-	return &serviceManager{cfg, filepath.Join(cfg.ConfigPath(), "state-svc.lock")}
+	return &serviceManager{cfg}
 }
 
 func (s *serviceManager) Start(args ...string) error {
-	curPid, err := s.Pid()
-	if err == nil && curPid != nil {
-		return errs.New("Service is already running")
-	}
+	var proc *os.Process
+	err := s.cfg.SetSafe(constants.SvcConfigPid, func(oldPidI interface{}) interface{} {
```

Comment:
```suggestion
	err := s.cfg.SetSafe(constants.SvcConfigPid, func(oldPidI interface{}) interface{} {
```

I like this approach, but I think putting this on `Set` is a bit confusing and limiting. What about a `cfg.WithLock(func() { ... })` ?

Also, I'd like to reserve the `unsafe` naming for things used in tests. `unsafe` should signal a big red flag and probably shouldn't be in production code.

Reply:
Another thought; maybe it's not appropriate to rely on the config locking mechanic here at all. The config locking should be an internal mechanic to locking. 

I think we should be trying to solve this in a way that doesn't require any special knowledge from code outside of the config package, though I recognize that's a bit out of scope for the current story. I'm ok landing this with a minimal changeset and filing a followup though.

Reply:
Yes, I am a little unsure about this:

We could certainly introduce a `state-svc.lock` lockfile for this, that just ensures that `state-svc` is never called twice, but IMO it is also okay to say that we need to ensure that we need a lock on the `constants.SvcConfigPid` key.  As this should not change between reading the old value and writing the new value.  Here, I opted for the latter and helped fix some other bugs :-).  But I could also be convinced that a second lock file is the way to go.